bundles/org.openhab.ui/web/src/components/sse-events-mixin.js

```diff
@@ -0,0 +1,154 @@
+      audioContext: null
+    }
+  },
+  methods: {
+    startEventSource () {
+      const topicAudio = 'openhab/webaudio/playurl'
+      const commandItem = localStorage.getItem('openhab.ui:commandItem')
+      const topicCommand = `openhab/items/${commandItem || ''}/command`
+      let topics = null
+      if (localStorage.getItem('openhab.ui:webaudio.enable') === 'enabled') {
+        topics = topicAudio
+      }
+      if (commandItem) {
+        topics = topics ? `${topics},${topicCommand}` : topicCommand
+      }
+      if (!topics) return
+      this.eventSource = this.$oh.sse.connect(`/rest/events?topics=${topics}`, null, (event) => {
+        console.debug('Received SSE event: ' + JSON.stringify(event))
+        switch (event.topic) {
+          case topicAudio:
+            const topicParts = event.topic.split('/')
+            switch (topicParts[2]) {
+              case 'playurl':
+                this.playAudioUrl(JSON.parse(event.payload))
+                break
+            }
+            break
+          case topicCommand:
+            const payload = JSON.parse(event.payload)
+            const [command, ...segments] = payload.value.trim().split(/(?<!\\):/)
+            const combined = segments.join(':')
+            switch (command) {
+              case 'navigate':
+                this.$f7.views.main.router.navigate(combined)
+                break
+              case 'popup':
+              case 'popover':
+              case 'sheet':
+                if (combined.indexOf('page:') !== 0 && combined.indexOf('widget:') !== 0 && combined.indexOf('oh-') !== 0) {
+                  console.error('Action target is not of the format page:uid or widget:uid')
+                  return
+                }
+                console.debug(`Opening ${combined} in ${command} modal`)
+                let modalRoute = {
+                  url: combined + '/' + command,
+                  route: {
+                  }
+                }
+                if (command === 'popup') modalRoute.route.popup = { component: OhPopup }
+                if (command === 'popover') modalRoute.route.popup = { component: OhPopover }
+                if (command === 'sheet') modalRoute.route.popup = { component: OhSheet }
+                let modalProps = {
+                  props: {
+                    uid: combined,
+                    modalParams: {}
+                  }
+                }
+                this.closePopups()
+                this.$f7.views.main.router.navigate(modalRoute, modalProps)
+                break
+              case 'close':
+                this.closePopups()
+                break
+              case 'back':
+                this.$f7.views.main.router.back()
+                break
+              case 'reload':
+                window.location.reload()
+                break
+              case 'notification':
+                const payload = {
+                  text: segments[0],
+                  closeOnClick: true,
```

Comment:
I would suggest to have a close button (`closeButton`) and swipe to close (`swipeToClose`), because both on iOS and Android a click on a notification does open it, but not close it.

This brings me to the next idea: I can imagine having the user define a action to perform when clicking on the notification. I would support the same actions as for the command Item.
This is BTW not directly supported, but if such a on-click action is defined, we could enable `closeOnClick` and add an event listener for closed, like here: https://framework7.io/docs/notification#examples.

I am not sure how to fit such an action into the `notification:Text:Title:Subtitle:Title_Right_Text:Close_Timeout` scheme, without making that totally unreadable. Probably something like `notification:Text...Closed_Timeout_navigate:/pages/page` (`_` as a delimiter?). Another option would be to have this parameterized, e.g. `notification:text=Some Text,title=Some Title,action=navigate:/pages/page`.

Reply:
`closeOnClick` can be removed IMO.

```suggestion

```